tests/integration_tests/allennlp_tests/test_allennlp.py

```diff
@@ -238,3 +238,58 @@ def objective(trial: optuna.Trial) -> float:
         study.optimize(objective, n_trials=1)
         assert study.trials[0].state == optuna.trial.TrialState.COMPLETE
         assert study.trials[0].value == 1.0
+
+
+def test_allennlp_pruning_callback_with_executor() -> None:
+    input_config_file = (
+        "tests/integration_tests/allennlp_tests/example_with_executor_and_pruner.jsonnet"
+    )
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        storage = "sqlite:///" + os.path.join(tmp_dir, "result.db")
+        serialization_dir = os.path.join(tmp_dir, "allennlp")
+
+        origin_pruner = optuna.pruners.HyperbandPruner(min_resource=3)
+        study = optuna.create_study(
+            direction="maximize",
+            pruner=origin_pruner,
+            storage=storage,
+        )
+        trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+        trial.suggest_float("DROPOUT", 0.0, 0.5)
+        executor = optuna.integration.AllenNLPExecutor(trial, input_config_file, serialization_dir)
+        executor.run()
+        target_pruner = optuna.integration.allennlp._create_pruner()
+
+        # It checks a type of target_pruner since mypy fail to check here.
+        # This is why `_create_pruner` returns `BasePruner`, which doesn't have
+        # _min_resource, _max_resource, and _reduction_factor that belongs to `HyperbandPruner`.
+        if isinstance(target_pruner, optuna.pruners.HyperbandPruner):
+            assert origin_pruner._min_resource == target_pruner._min_resource
+            assert origin_pruner._max_resource == target_pruner._max_resource
+            assert origin_pruner._reduction_factor == target_pruner._reduction_factor
+        else:
+            assert False  # Always fail
+
+
+def test_infer_and_cast() -> None:
+    assert optuna.integration.allennlp._infer_and_cast(None) is None
+    assert optuna.integration.allennlp._infer_and_cast("True") is True
+    assert optuna.integration.allennlp._infer_and_cast("False") is False
+    assert optuna.integration.allennlp._infer_and_cast("3.14") == 3.14
+    assert optuna.integration.allennlp._infer_and_cast("42") == 42
+    assert optuna.integration.allennlp._infer_and_cast("auto") == "auto"
+
+
+def test_fetch_pruner_config() -> None:
+    class SomeNewPruner(optuna.pruners.BasePruner):
+        def __init__(self) -> None:
+            pass
+
+        def prune(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> bool:
+            return False
+
+    study = optuna.create_study(pruner=SomeNewPruner())
+    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+
+    with pytest.raises(ValueError):
+        optuna.integration.allennlp._fetch_pruner_config(trial)
```

Comment:
Could you add a test case to check if `_fetch_pruner_config` does not raise errors when it receives supported pruners?

Reply:
In 07a9de9, I added a parametrized test. In this test, I found some bugs. Thank you so much for the suggestion.